soc/nordic/common/dmm.c

```diff
@@ -295,3 +295,10 @@ int dmm_init(void)
 
 	return 0;
 }
+
+/*
+ * dmm_init relies on kheap, initialized in PRE_KERNEL_1 at
+ * CONFIG_KERNEL_INIT_PRIORITY_OBJECTS. To enable early device use of
+ * dmm, we initialize immediately after kheap.
+ */
+SYS_INIT(dmm_init, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
```

Comment:
Now `dmm_init()` does not need to be externally accessible, so it can be made static and the changes in `dmm.h` from c84c2fc37dd0d76c5b6735ed72c25de073ea2993 can be reverted.

Reply:
fixed in https://github.com/zephyrproject-rtos/zephyr/pull/89333